### PR TITLE
change initialization order of CSI and cameras

### PR DESCRIPTION
**Pull request by bitgamma**
`video_mcux_csi_init`, which setups the CSI pins (i.e: calls pinctrl_apply_state) was called after `mt9m114_init` which tries to do i2c communication with the camera to read the chip id. But since one of the CSI pins is the camera master clock, doing things in this order won't work. This PR inverts the order in which the devices are initialized.

also, somewhat related, correct the PWDN pin for pinmux_csi

### CODE REVIEW

File: boards/arm/mimxrt1064_evk/mimxrt1064_evk-pinctrl.dtsi [hunk 1]
```diff
@@ -24,7 +24,7 @@
 	/* conflicts with lpuart3 and flexcan1 */
 	pinmux_csi: pinmux_csi {
 		group0 {
-			pinmux = <&iomuxc_gpio_ad_b0_04_gpio1_io04>;
+			pinmux = <&iomuxc_gpio_ad_b1_02_gpio1_io18>;
```
Comment (loicpoulain) [inline boards/arm/mimxrt1064_evk/mimxrt1064_evk-pinctrl.dtsi:27]: should be in a different commit.

Comment (bitgamma) [reply boards/arm/mimxrt1064_evk/mimxrt1064_evk-pinctrl.dtsi:27]: done

**Review by loicpoulain — COMMENTED** [review]
Can you please add a commit message explaining the change here.

**Comment by bitgamma** [comment]
> Can you please add a commit message explaining the change here.

done!

**Comment by bitgamma** [comment]
any chance to get this merged?

**Comment by teburd** [comment]
@loicpoulain please re-review when you get a chance, thanks!

**Review by loicpoulain — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]

File: drivers/video/video_mcux_csi.c [hunk 2]
```diff
@@ -443,6 +445,6 @@ static int video_mcux_csi_init_0(const struct device *dev)
 DEVICE_DT_INST_DEFINE(0, &video_mcux_csi_init_0,
 		    NULL, &video_mcux_csi_data_0,
 		    &video_mcux_csi_config_0,
-		    POST_KERNEL, CONFIG_VIDEO_INIT_PRIORITY,
+		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
```
Comment (dleach02) [inline drivers/video/video_mcux_csi.c:448]: I would add a comment here to highlight the dependency. Otherwise, future folks may not catch the dependency.

**Comment by dleach02** [comment]
@bitgamma please address the compliance issues with your commit message. Look at the history on these files for guidance.

File: drivers/video/video_mcux_csi.c (hunk 2 above)
Comment (bitgamma) [reply drivers/video/video_mcux_csi.c:453]: done

**Comment by bitgamma** [comment]
> @bitgamma please address the compliance issues with your commit message. Look at the history on these files for guidance.

done, thanks for the review

File: boards/arm/mimxrt1064_evk/mimxrt1064_evk-pinctrl.dtsi [hunk 3]
```diff
@@ -24,7 +24,7 @@
 	/* conflicts with lpuart3 and flexcan1 */
 	pinmux_csi: pinmux_csi {
 		group0 {
-			pinmux = <&iomuxc_gpio_ad_b0_04_gpio1_io04>;
```
Comment (danieldegrasse) [inline boards/arm/mimxrt1064_evk/mimxrt1064_evk-pinctrl.dtsi:27]: Hi @bitgamma, can you update this commit title? Something like the following:

`boards: arm: mimxrt1064_evk: correct PWDN pin of the CSI interface`

Apologies for not catching this earlier

Comment (bitgamma) [reply boards/arm/mimxrt1064_evk/mimxrt1064_evk-pinctrl.dtsi:27]: done

**Review by mmahadevan108 — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@dleach02 @danieldegrasse, can you take a look

**Review by dleach02 — APPROVED** [review]
